WidgetExtension/Variants/FollowersCount/FollowersCountWidget.swift

```diff
@@ -0,0 +1,149 @@
+
+struct FollowersCountWidgetProvider: IntentTimelineProvider {
+    private let followersHistory = FollowersCountHistory.shared
+    
+    func placeholder(in context: Context) -> FollowersCountEntry {
+        .placeholder
+    }
+
+    func getSnapshot(for configuration: FollowersCountIntent, in context: Context, completion: @escaping (FollowersCountEntry) -> ()) {
+        loadCurrentEntry(for: configuration, in: context, completion: completion)
+    }
+
+    func getTimeline(for configuration: FollowersCountIntent, in context: Context, completion: @escaping (Timeline<FollowersCountEntry>) -> ()) {
+        loadCurrentEntry(for: configuration, in: context) { entry in
+            completion(Timeline(entries: [entry], policy: .after(.now)))
+        }
+    }
+}
+
+struct FollowersCountEntry: TimelineEntry {
+    let date: Date
+    let account: FollowersEntryAccountable?
+    let configuration: FollowersCountIntent
+    
+    static var placeholder: Self {
+        FollowersCountEntry(
+            date: .now,
+            account: FollowersEntryAccount(
+                followersCount: 99_900,
+                displayNameWithFallback: "Mastodon",
+                acct: "mastodon",
+                avatarImage: UIImage(named: "missingAvatar")!,
+                domain: "mastodon"
+            ),
+            configuration: FollowersCountIntent()
+        )
+    }
+    
+    static var unconfigured: Self {
+        FollowersCountEntry(
+            date: .now,
+            account: nil,
+            configuration: FollowersCountIntent()
+        )
+    }
+}
+
+struct FollowersCountWidget: Widget {
+    private var availableFamilies: [WidgetFamily] {
+        if #available(iOS 16, *) {
+            return [.systemSmall, .accessoryRectangular, .accessoryCircular]
+        }
+        return [.systemSmall]
+    }
+
+    var body: some WidgetConfiguration {
+        IntentConfiguration(kind: "Followers", intent: FollowersCountIntent.self, provider: FollowersCountWidgetProvider()) { entry in
+            FollowersCountWidgetView(entry: entry)
+        }
+        .configurationDisplayName("Followers")
+        .description("Show number of followers.")
+        .supportedFamilies(availableFamilies)
+    }
+}
+
+private extension FollowersCountWidgetProvider {
+    func loadCurrentEntry(for configuration: FollowersCountIntent, in context: Context, completion: @escaping (FollowersCountEntry) -> Void) {
+        Task {
+            guard
+                let authBox = WidgetExtension.appContext
+                    .authenticationService
+                    .mastodonAuthenticationBoxes
+                    .first
+            else {
+                guard !context.isPreview else {
+                    return completion(.placeholder)
+                }
+                return completion(.unconfigured)
+            }
+            
+            guard let desiredAccount: String = {
+                guard let account = configuration.account else {
```

Comment:
I was a bit confused by this nested `guard` with a function :D

Reply:
I'll refactor this 😄